vm/native/native.cpp

```diff
@@ -69,7 +70,9 @@ extern "C" {
   }
 
   uintptr_t gcMallocAtomic(int32_t size) {
-    return (uintptr_t)GC_MALLOC_ATOMIC(size);
+    uintptr_t ptr = (uintptr_t)GC_MALLOC_ATOMIC(size);
+    memset((void*)ptr, 0, size);
```

Comment:
Nit: we could subtract the size of the header here because we don't need to zero it out, but I guess it doesn't make a big difference.